kie-wb-webapp-test/pom.xml

```diff
@@ -0,0 +1,164 @@
+                            </artifactItem>
+                        </artifactItems>
+                         <jvm-args>-XX:MaxPermSize=2048m -XX:PermSize=2048m</jvm-args>
+                          <systemProperties>
+                            <JAVA_OPTS>-Xmx3000m -Xms1000m -XX:PermSize=2000m -XX:MaxPermSize=4000m</JAVA_OPTS>
+                          </systemProperties>
+                    </configuration>
+                </execution>
+            </executions>
+        </plugin>
+        <!-- Copy server configuration -->
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-resources-plugin</artifactId>
+            <version>${version.plugin.resources}</version>
+            <executions>
+                <execution>
+                    <id>copy-standalone-xml</id>
+                    <phase>generate-test-sources</phase>
+                    <goals>
+                        <goal>copy-resources</goal>
+                    </goals>
+                    <configuration>
+                        <outputDirectory>${jboss.home}/standalone/configuration</outputDirectory>
+                        <resources>
+                            <resource>
+                                <directory>../kie-wb/kie-wb-webapp/src/main/resources</directory>
+                            </resource>
+                        </resources>
+                         <jvm-args>-XX:MaxPermSize=2048m -XX:PermSize=2048m</jvm-args>
+                         <systemProperties>
+                            <JAVA_OPTS>-Xmx3000m -Xms1000m -XX:PermSize=2000m -XX:MaxPermSize=4000m</JAVA_OPTS>
+                          </systemProperties>
+                    </configuration>
+                </execution>
+            </executions>
+        </plugin>
+        <!-- WildFly plugin to deploy war -->
+        <plugin>
+            <groupId>org.wildfly.plugins</groupId>
+            <artifactId>wildfly-maven-plugin</artifactId>
+            <version>${version.wildfly.maven.plugin}</version>
+            <configuration>
+                <jbossHome>${jboss.home}</jbossHome>
+              <javaOpts>
+                <javaOpt>-Xms2048m</javaOpt>
+                <javaOpt>-Xmx4096m</javaOpt>
+                <javaOpt>-XX:MaxPermSize=2048m</javaOpt>
+              </javaOpts>
+
+                <serverConfig>standalone-full.xml</serverConfig>
+                <targetDir>../</targetDir>
+                <filename>/kie-wb/kie-wb-distribution-wars/target/kie-wb-6.3.0-SNAPSHOT-wildfly8.war</filename>
+            </configuration>
+            <executions>
+                <execution>
+                    <id>pre-integration-phase</id>
+                    <phase>pre-integration-test</phase>
+                    <goals>
+                        <goal>start</goal>
+                        <goal>deploy</goal>
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
```

Comment:
This looks like a perfect use case for cargo (codehaus-cargo.github.io). The lines `48` to `133` could be replaced by the following cargo config:
```xml
<plugin>
  <groupId>org.codehaus.cargo</groupId>
  <artifactId>cargo-maven2-plugin</artifactId>
  <configuration>
    <container>
      <containerId>wildfly8x</containerId>
      <type>installed</type>
      <zipUrlInstaller>
        <url>${wildfly8x.download.url}</url>
      </zipUrlInstaller>
    </container>
    <configuration>
      <properties>
        <cargo.jboss.configuration>standalone-full</cargo.jboss.configuration>
      </properties>
    </configuration>
    <deployables>
      <deployable>
        <groupId>org.kie</groupId>
        <artifactId>kie-wb-distribution-wars</artifactId>
        <classifier>wildfly8</classifier>
        <type>war</type>
      </deployable>
    </deployables>
  </configuration>
</plugin>
```
The advantage is of this approach is that we are not tied to any particular container. We can create different profiles for different containers and execute the tests easily on Tomcats, WildFly and EAP.